nearcore/src/config.rs

```diff
@@ -217,6 +231,7 @@ pub struct Config {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub save_trie_changes: Option<bool>,
     pub log_summary_style: LogSummaryStyle,
+    #[serde(with = "near_async::time::serde_duration_as_std")]
```

Comment:
curious why do we need to define this? Can we not have a default serde serialization/deserialization function?

Reply:
As per this article https://serde.rs/custom-serialization.html seems like we wouldn't need to define the `serde_duration_as_std` module but can just shift the trait impl to outside the module.

I think once we have this, we wouldn't need to define custom serialize/deserialize for Option<Duration> either?

Reply:
Hmm, damn... Ok, this scares me as it would be very easy for people to use the default serialize/deserialize for Duration instead of `serde_duration_as_std`? I don't have a solution to this. Maybe we can reimplement the default serialize/deserialize for Duration and disable `serde` feature so that we explicitly need to choose which implementation of serialize/deserialize we want? Or is that also not possible?

Reply:
```
// TODO: consider wrapping these types to prevent interactions
// with other time libraries, especially to prevent the direct access
// to the realtime (i.e. not through the Clock).
```
Might this be a good time to also consider wrapping time primitives?

Reply:
> Hmm, damn... Ok, this scares me as it would be very easy for people to use the default serialize/deserialize for Duration instead of serde_duration_as_std?

That's why I added this dreaded test in config_duration_test.rs.

Wrapping the primitives is an option, but ... it's... so not cool. I'm not opposed to it, it just sounds a bit like a headache and I'm afraid of https://xkcd.com/927/ too